stat.go

```diff
@@ -648,6 +648,71 @@ func KullbackLeibler(p, q []float64) float64 {
 	return kl
 }
 
+// OrdinaryLinearRegression computes the simple linear regression of x and y.
+// It returns the values for \hat{\alpha}, \hat{\beta} and the coefficient
+// of determination, r. If origin is true, the regression is forced to pass
+// through the origin.
+// The lengths of x and y must be equal. If weights is nil then all of the
+// weights are 1. If weights is not nil, then len(x) must equal len(weights).
```

Comment:
a) OK. Note that the calculation changes dependent on whether the regression is through the origin or not.

b) There is a duplication of the Mean call in the case that the regression is not through the origin.

d) It is given prominence in R. Normally I would not defer to other languages, but R is the de facto standard for statistical computing, backed by many statisticians.

*If it were* added, it would need to have a weights parameter. The meaning of this is difficult.